packages/syft/src/syft/service/action/action_object.py

```diff
@@ -801,17 +793,25 @@ def reload_cache(self) -> SyftError | None:
 
         return None
 
-    def _save_to_blob_storage_(self, data: Any) -> SyftError | None:
+    def _save_to_blob_storage_(self, data: Any) -> SyftError | SyftWarning | None:
         # relative
         from ...types.blob_storage import BlobFile
         from ...types.blob_storage import CreateBlobStorageEntry
 
+        api = APIRegistry.api_for(self.syft_node_location, self.syft_client_verify_key)
+        if api is None:
+            raise ValueError(
+                f"api is None. You must login to {self.syft_node_location}"
+            )
+        if not can_upload_to_blob_storage(data, api.metadata):
```

Comment:
This check should be move to else statement line:816. Since for BlobFile we have a different flow and for ActionDataEmpty we don't want to set any values.